test/json_product_generation_test.dart

```diff
@@ -22,8 +23,20 @@ void main() {
     expect(result.status, 1);
     Product product = result.product;
     Map<String, dynamic> productMap = product.toJson();
-    print(JsonHelper.selectedImagesToJson(product.selectedImages));
     String json = jsonEncode(productMap);
     assert(json is String);
   });
-}
\ No newline at end of file
+
+  test('EcoscoreData JSON generation', () {
+    final product = Product();
+    product.productName = 'TestProduct';
```

Comment:
For the readability I would suggest to use the cascade operator (see https://dart.dev/guides/language/language-tour#cascade-notation- )

Reply:
I cleaned up the test. I was able to simply pass the field values via constructors.